### DISCUSSION TITLE
Enable/Disable Safe Search for specific entry per client

### DISCUSSION THREAD

**Discussion by abishur** [AdguardTeam/AdGuardHome#4715] [Q&A]
Would it be possible to select which safe search options are enabled for each individual client?

For example, globally I might want to enforce a safe search for google or duckduckgo, but in Youtube I would only want certain clients (kid devices) to browse in restricted mode.

Currently it looks like it's an all or nothing deal.  Admittedly, I just started using adguard home in the past hour so it could be I'm missing something, lol

**Comment by fernvenue** [comment]
Hi abishur, as you can see, this is an ongoing known issue: https://github.com/AdguardTeam/AdGuardHome/issues/1163. For now, you can do it by Custom filtering rules. As in your example:

> For example, globally I might want to enforce a safe search for google or duckduckgo, but in Youtube I would only want certain clients (kid devices) to browse in restricted mode.

Assuming that your device's IP address is `10.0.0.1`, then we can disable YouTube Safe Search for this device by adding the following rules to the Custom filtering rules:

```
@@/(m|www).youtube.com/$important,client="10.0.0.1"
@@/(youtube|youtubei).googleapis.com/$important,client="10.0.0.1"
@@www.youtube-nocookie.com$important,client="10.0.0.1"
```

Of course, in the actual use environment, we may have multiple devices, and the devices may increase or decrease in the future. For this case, we can also use [ctag](https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#ctag):

```
@@/(m|www).youtube.com/$important,ctag=user_regular
@@/(youtube|youtubei).googleapis.com/$important,ctag=user_regular
@@www.youtube-nocookie.com$important,ctag=user_regular
```

Here, the extra work you need to do is add a tag called `user_regular` for these adult devices, if you don't know how to do that just follow the AdGuardHome Wiki: [Clients](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients).

Here, the regular matching YouTube domains are all from the source code of AdGuardHome:

https://github.com/AdguardTeam/AdGuardHome/blob/b3210cfa7ea079676a877f6aaea3fc6210dfe286/internal/filtering/safesearch.go#L374-L378

You can find domains for other services above and below the code above, hope this solves your current problem and hopefully this issue will move forward sooner :)

**Reply by abishur** [reply]
I completely missed this when I searched prior to opening a new discussion 🤦‍♂️
Is there a way to add a new ctag category of device_adult?  I didn't see it listed as an option and I didn't see anything referencing on how to add custom ctags.

**Reply by fernvenue** [reply]
> I completely missed this when I searched prior to opening a new discussion 🤦‍♂️ Is there a way to add a new ctag category of device_adult? I didn't see it listed as an option and I didn't see anything referencing on how to add custom ctags.

Yes, AdGuardHome doesn't currently have custom tags to use, and that's just an example above, not stating that it's just an example is my mistake. These are the user tags that AdGuardHome currently offers:

- `user_admin`: administrators.
- `user_regular`: regular users.
- `user_child`: children.

You can find them on Web-based GUI, or you may want to read the [documentation](https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#ctag) first :)

Thanks for pointing this out, as the fictitious `device_adult` example above may mislead future users viewing this discussion, I edited and used the actual tag `user_regular` currently provided by AdGuardHome.

**Comment by abishur** [comment]
Thanks! This is what I entered into my custom filter list to get it to work:

```
@@||(m|www).youtube.com/$important,ctag=user_admin
@@||(youtube|youtubei).googleapis.com$important,ctag=user_admin
@@||www.youtube-nocookie.com$important,ctag=user_admin
```
Though I think I will take a look at that documentation before I move it out of testing 🤣

**Reply by fernvenue** [reply]
Haha, I don't think this gonna work, because you should use `/.../` for regex.

**Reply by abishur** [reply]
For some reason when I do the regex it still puts me in restricted mode

**Reply by fernvenue** [reply]
> For some reason when I do the regex it still puts me in restricted mode

Maybe you can find the specific reason from the query log or verbose log, see if AdGuardHome actually receives requests and how it handles them. I tend to think it's a cache or rule matching error.

**Comment by Unknown209ED** [comment]
How do i fix my fireguard code it is supposed to be a unstoppable safe search but the word "pornography" gets though
